### PR TITLE
get_ghost() cleanup

### PR DESCRIPTION
**Pull request by Rohesie**
Micro cleanup. Started it because of #2184 but then saw the issue was a bit deeper.

### CODE REVIEW

File: code/game/objects/items/defibrillator.dm [hunk 1]
```diff
@@ -66,14 +66,18 @@
...
+
 /mob/living/carbon/human/proc/get_ghost()
-	if(mind && !client) //Let's call up the correct ghost! Also, bodies with clients only, thank you.
-		for(var/mob/dead/observer/G in GLOB.player_list)
-			if(G.mind == mind)
-				var/mob/dead/observer/ghost = G
-				if(ghost && ghost.client && ghost.can_reenter_corpse)
-					return ghost
-	return 0
+	if(!mind || client) //Let's call up the correct ghost!
```
Comment (DominikPanic) [inline code/game/objects/items/defibrillator.dm]: See my comment on the bug report, this won't work.

Comment (Rohesie) [reply code/game/objects/items/defibrillator.dm]: Yep, the issue is deeper. Was waiting on your input. This PR only refactors the currently non-working proc.

The problem is the lack of a reference to the ghost in the mob at the moment.
Suggestions?

Comment (DominikPanic) [reply code/game/objects/items/defibrillator.dm]: Like I said. remove the mind check, check if ghost.current == src

Comment (DominikPanic) [inline code/game/objects/items/defibrillator.dm]:
```suggestion
	if(client) //Let's call up the correct ghost!
```

File: code/game/objects/items/defibrillator.dm [hunk 2]
```diff
@@ -66,14 +66,18 @@
...
+		return FALSE
+	for(var/g in GLOB.observer_list)
+		var/mob/dead/observer/ghost = g
+		if(ghost.mind != mind)
```
Comment (DominikPanic) [inline code/game/objects/items/defibrillator.dm]:
```suggestion
		if(ghost.current != src)
```

File: code/game/objects/items/defibrillator.dm [hunk 3]
```diff
@@ -66,14 +66,18 @@
...
-				if(ghost && ghost.client && ghost.can_reenter_corpse)
-					return ghost
-	return 0
+	if(client) //Let's call up the correct ghost!
+		return FALSE
+	for(var/g in GLOB.observer_list)
+		var/mob/dead/observer/ghost = g
+		if(ghost.mind.current != src)
+			continue
+		if(ghost.client && ghost.can_reenter_corpse)
```
Comment (psykzz) [inline code/game/objects/items/defibrillator.dm]: does it matter to check the if the clients are the same?

Comment (DominikPanic) [reply code/game/objects/items/defibrillator.dm]: There is no check for clients being the same?

Comment (psykzz) [reply code/game/objects/items/defibrillator.dm]: You are right, I was being dense. 
Ignore me :)

**Pull request merged**